app/models/soda.rb

```diff
@@ -0,0 +1,18 @@
+class Soda < ApplicationRecord
+  belongs_to :creator, class_name: 'User'
+  belongs_to :last_editor, class_name: 'User'
+  belongs_to :country
+  has_many :photos
+  has_and_belongs_to_many :flavors
```

Comment:
I think you are proposing a single level of flavors, is that right? Like there is no flavor "strawberry limeade", but there is "strawberry", "lime", and "lemon"? This is definitely simpler in terms of a data model, but it also has other costs.

The original reason I suggested compound flavors (that's my new name for it) is to avoid the need to add every tag individually. In this way, we reduce the amount of work a person adding a soda needs to do (some tags get added automatically), and also has the potential to reduce user error.

So let's take "appelsin" - it's a type of soda which is orange flavored, but it is not a brand. "Appelsin" is a concept. So every "appelsin" is "orange" flavored, and every "appelsin" soda will need to add both of those tags. It introduces the potential for user error, where they only choose "orange" or "appelsin" but not both, and we could lose data that "appelsins" are always "orange", so that searching "orange sodas" might not return appelsins if they were incorrectly tagged. 

"Where the data that "appelsin" is orange will come from?" - good question. I was planning to pre-seed the db with most of the possible flavors of sodas I (and the internet) can imagine, and set up those relationships first. None of our plans currently involve a way for users to set up those relationships, but I think that's ok. I am imagining that new soda flavors will be rarely added to the db, since 99% of sodas will be covered by the standard flavors. I mentioned "cherry limeade" because it is a common flavor - several soda companies produce this flavor. But something like "acai bluberry fizz" is going to be pretty rare, so that soda is good to just get tagged "acai" and "blueberry". I think originally I was thinking the value was with flavors that do not represent what they are supposed to taste like. This brings me to another point - 

I think we should define flavors as "flavor notes" and not "the flavor the soda says it is". For example, there is a rocket fizz soda called "London Fog" which is a cream soda. I would argue that "London Fog" is not actually the "flavor" - that is the name, and the flavor is "cream". However, I can see a user being confused and trying to enter "London Fog" into the tag box. I propose two options to deal with this:

1. label the field "flavor notes" or possibly "flavor notes/types" so that appelsin and orange both still make sense in that field

2. split out sodas into a hierarchy. I imagine something like:

Cola
Ginger Ale
Root Beer
National sodas
Fruit or vegetable

actually nevermind, I already see the problem here: these are not exclusive. "cherry coke" is both "cherry" and "cola" flavored, so this is a bad idea. 

We have discussed that we basically already need to curate flavors - we want an alert when people add new ones, because we don't expect them to be doing that very often.

"I was thinking may be we can create in the notes field: add flavors to soda." - can you clarify what this means? 

"We also can create a field: complete_flavor and store a string that is a sum of those tags in the specific order the user entered it if needed" - what would the value of this field be? Where would we use it?